src/main/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateValidateModelStep.java

```diff
@@ -32,22 +36,88 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     Dataset dataset = datasetService.retrieve(datasetId);
     List<String> existingTableNames =
         dataset.getTables().stream().map(DatasetTable::getName).collect(Collectors.toList());
-    List<String> newTableNames = DatasetSchemaUpdateUtils.getNewTableNames(updateModel);
-    Collection<String> uniqueTableNames =
-        CollectionUtils.intersection(existingTableNames, newTableNames);
-
-    if (!uniqueTableNames.isEmpty()) {
-      return new StepResult(
-          StepStatus.STEP_RESULT_FAILURE_FATAL,
-          new DatasetSchemaUpdateException(
-              "Could not validate table additions",
-              List.of(
-                  "Found new tables that would overwrite existing tables",
-                  String.join(", ", uniqueTableNames))));
+    final List<String> newTableNames;
+    if (DatasetSchemaUpdateUtils.hasTableAdditions(updateModel)) {
+      newTableNames = DatasetSchemaUpdateUtils.getNewTableNames(updateModel);
+      List<String> uniqueTableNames = ListUtils.intersection(existingTableNames, newTableNames);
+      if (!uniqueTableNames.isEmpty()) {
+        return failsValidation(
+            "Could not validate table additions",
+            List.of(
+                "Found new tables that would overwrite existing tables",
+                String.join(", ", uniqueTableNames)));
+      }
+    } else {
+      newTableNames = List.of();
     }
+
+    if (DatasetSchemaUpdateUtils.hasColumnAdditions(updateModel)) {
+      List<DatasetSchemaColumnUpdateModel> addColumns = updateModel.getChanges().getAddColumns();
+      List<String> newAndExistingTableNames = ListUtils.union(existingTableNames, newTableNames);
```

Comment:
The only use of this object is to call `contains()`. Given that, it seems like a `Set` would make more sense here.
```suggestion
      Set<String> allTableNames = new HashSet<>(existingTableNames);
      allTableNames.addAll(newTableNames);
```